src/cases/models/case.js

```diff
@@ -39,18 +40,63 @@ export class Case {
     });
   }
 
-  assignUser({ assignedUserId, text, createdBy }) {
+  setAssignedUser(userId) {
+    this.assignedUserId = userId;
+    this.assignedUser = userId ? { id: userId } : null;
+  }
+
+  findTask(stageId, taskGroupId, taskId) {
+    const stage = this.stages.find((s) => s.id === stageId);
+    const taskGroup = stage?.taskGroups.find((tg) => tg.id === taskGroupId);
+    const task = taskGroup?.tasks.find((t) => t.id === taskId);
+
+    if (!task) {
+      throw Boom.notFound(
+        `Can not find Task with id ${taskId} from taskGroup ${taskGroupId} in stage ${stageId}`,
+      );
+    }
+
+    return task;
+  }
+
+  updateTaskStatus(stageId, taskGroupId, taskId, status, note) {
+    const authenticatedUserId = getAuthenticatedUser().id;
```

Comment:
See comment on import of `getAuthenticatedUser` above, we shouldn't do this in the domain model